src/core/components/InstallButton/index.js

```diff
@@ -194,195 +185,148 @@ export class InstallButtonBase extends React.Component {
-    // Firefox" button regardless on mozAddonManager support is a better UX.
-    const useButton =
-      (hasAddonManager !== undefined && !hasAddonManager) ||
-      addon.type === ADDON_TYPE_OPENSEARCH ||
-      this.props.useButton;
-    let button;
-
-    const { compatible } = getClientCompatibility({
-      addon,
-      clientApp,
-      userAgentInfo,
-    });
-
-    const buttonIsDisabled = !compatible;
-    const buttonClass = makeClassName(
-      'InstallButton-button',
-      'Button--action',
-      className,
-      {
-        'InstallButton-button--disabled': buttonIsDisabled,
-      },
-    );
     const installURL = findInstallURL({
       defaultInstallSource,
       location,
       platformFiles: addon.platformFiles,
       userAgentInfo,
     });
 
-    if (addon.type === ADDON_TYPE_THEME) {
-      button = (
-        <Button
-          buttonType="action"
-          className={buttonClass}
-          disabled={buttonIsDisabled}
-          data-browsertheme={JSON.stringify(getThemeData(addon))}
-          onClick={this.installTheme}
-          puffy
-        >
-          <Icon name="plus" />
-          {i18n.gettext('Install Theme')}
-        </Button>
-      );
-    } else if (addon.type === ADDON_TYPE_OPENSEARCH) {
-      const onClick = buttonIsDisabled
-        ? null
-        : (event) => {
-            event.preventDefault();
-            event.stopPropagation();
-
-            _log.info('Adding OpenSearch Provider', { addon });
-            _window.external.AddSearchProvider(installURL);
-            this.trackInstallStarted({
-              addonName: addon.name,
-              type: addon.type,
-            });
-
-            return false;
-          };
-      button = (
-        <Button
-          buttonType="action"
-          className={buttonClass}
-          disabled={buttonIsDisabled}
-          onClick={onClick}
-          href={installURL}
-          prependClientApp={false}
-          prependLang={false}
-          puffy
-        >
-          <Icon name="plus" />
-          {i18n.gettext('Add to Firefox')}
-        </Button>
-      );
-    } else {
-      const onClick = buttonIsDisabled
-        ? (event) => {
-            event.preventDefault();
-            event.stopPropagation();
-            return false;
-          }
-        : (event) => {
-            this.installExtension({ event, installURL });
-          };
-
-      const buttonText = isTheme(addon.type)
-        ? i18n.gettext('Install Theme')
-        : i18n.gettext('Add to Firefox');
-
-      button = (
-        <Button
-          buttonType="action"
-          className={buttonClass}
-          disabled={buttonIsDisabled}
-          onClick={onClick}
-          href={installURL}
-          prependClientApp={false}
-          prependLang={false}
-          puffy
-        >
-          <Icon name="plus" />
-          {buttonText}
-        </Button>
-      );
+    const buttonIsDisabled = disabled === false ? status === UNKNOWN : disabled;
```

Comment:
How and why would one want to use the mozAddonManager? If one wants to add add-ons, requiring mozAddonManager makes sense, no? Isn't it the new/recommended API? 

Which fallback should we use otherwise? The very old InstallTrigger? I checked and mozAddonManager is available in all supported FF versions.

Reply:
i think the button still contains the link to the theme, so a user could probably download the theme and install it manually, no? Could it be a valid fallback?

Reply:
In addition, currently AMO does not use mozAddonManager at all IIRC, so that's why -dev shows an enabled button. What would be the result with the disco app though?